io-engine-tests/src/lib.rs

```diff
@@ -188,6 +188,28 @@ pub fn truncate_file_bytes(path: &str, size: u64) {
     assert!(output.status.success());
 }
 
+/// Automatically assign a loopdev to path
+pub fn setup_loopdev_file(path: &str, sector_size: Option<u64>) -> String {
```

Comment:
We don't need a loopdevice, we should keep uses of aio/uring to fallocated file as we want to potentially make it more robust and use it for prod replication for certain use cases

Reply:
> We don't need a loopdevice, we should keep uses of aio/uring to fallocated file as we want to potentially make it more robust and use it for prod replication for certain use cases

It's a helper added for this case and potentially any future uses. With fallocated files we can't simulate larger sector size, but loopdev allows us to do that.

Reply:
IIRC at some point I wanted to add something like this to test helpers, but ended up with some other solution.

Reply:
> > We don't need a loopdevice, we should keep uses of aio/uring to fallocated file as we want to potentially make it more robust and use it for prod replication for certain use cases
> 
> It's a helper added for this case and potentially any future uses. With fallocated files we can't simulate larger sector size, but loopdev allows us to do that.

Ah I see :)